spinalcordtoolbox/deepseg/inference.py

```diff
@@ -126,12 +126,14 @@ def segment_non_ivadomed(path_model, model_type, input_filenames, threshold, kee
         fnames_out, targets = inference(path_img=fname_in, tmpdir=tmpdir, predictor=net, device=device)
         for fname_out, target in zip(fnames_out, targets):
             im_out = Image(fname_out)
+            # add soft threshold (not binarization)
+            im_out.data[im_out.data < 0.1] = 0
             # Apply postprocessing (replicates existing functionality from ivadomed package)
             # 1. Binarize predictions based on the threshold value
             # NOTE: Any post-processing is done only after thresholding (as post-processing operations
             # expect binary masks)
-            if threshold is not None:
-                im_out.data = binarize(im_out.data, threshold)
+            # if threshold is not None:
+            #     im_out.data = binarize(im_out.data, threshold)
```

Comment:
Note: The "hacky" fix in aadf57aa587f6cd921f0946514abb2c3a19e9d71 is superseded by https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4508 (which adds a soft thresholding for just the MONAI models).

Once that PR is merged, we should be able to remove the commit in this PR. :)

Reply:
Actually, scratch that. I'll merge this one first, then test softsegs in the other PR. :)